### DISCUSSION TITLE
How serverpod handles the concurrent connections?

### DISCUSSION THREAD

**Discussion by gintominto5329** [serverpod/serverpod#326] [Q&A] [answered]
Hello,

We are currently developing an app in flutter, and considering to use dart on the backend too,

So may i ask,
how does it handle multiple concurrent connection, 
also how much,
as i have heard, that multi threading on per-connection basis, is not possible in dart

Thanking you...

**Comment by vlidholt** [comment, marked as answer]
Dart can easily handle concurrent connections. It's true that the main code is running in a single thread, but under the hood the connections are handled in separate threads (in futures). The only issue is that if you are running Serverpod on a server with many CPUs it may not utilize them efficiently. This can be solved by running two (or more) servers on a single machine.

**Reply by kieran-myopnotes** [reply]
What about if you don't want to allow concurrent connections for security reasons, or you want to limit or at least report on it in some way (perhaps show the user all their current sessions with a way to kick another device / session off)

**Reply by gintominto5329** [reply]
> What about if you don't want to allow concurrent connections for security reasons, or you want to limit or at least report on it in some way (perhaps show the user all their current sessions with a way to kick another device / session off) 
> 

Are you a beginner?
if ,then just create a variable(with user's id) ,either global ,or with state-management ,and send error if attempted otherwise

**Reply by kieran-myopnotes** [reply]
No I am not a beginner

I am talking about a server-side method of detecting concurrent sessions for the same user on different devices and then managing them. Like when you check your current sessions on something like Google and you can see all the different devices, or at the very least device or IP information for all the various places you are currently signed into. Considering serverpod has built-in session management I was wondering if there was some way to manage all the sessions for a user. It appears there is not so I will add a feature request for this
